agent.py

```diff
@@ -1 +1,95 @@
 # agent.py
+
+# agent_type can be either "READING_AGENT" | "WRITING_AGENT" | "BROWSING_AGENT" | "CUSTOM_AGENT"
+# Create enum for agent_type and assign to agent argument agent_type
+import time
+from typing import Optional
+
+from agent_utils import (
+    AgentRunResult,
+    AgentType,
+    Model,
+    ModelProvider,
+    PermissionLabels,
+    PermissionMapping,
+    get_system_prompt,
+    llm_finished_keyword,
+    validate_agent_type_and_folder_to_mount,
+    validate_agent_type_and_permission,
+    validate_agent_type_and_system_prompt,
+)
+from llm.openai_api import OpenAIResponseApi
+
+tools = []
+
+
+class Agent:
+
+    def __init__(
+        self,
+        agent_type: AgentType,
+        model_details: Model,
+        system_prompt: Optional[str] = None,
+        environment: Optional[any] = None,  # TODO: Need to pass environment class
+        additional_tools: Optional[str] = None,
+        folder_to_mount: Optional[str] = None,
+        permission: Optional[PermissionLabels] = None,
+    ):
+        # validate init values before assigning
+        validate_agent_type_and_system_prompt(agent_type, system_prompt)
+        validate_agent_type_and_permission(agent_type, permission)
+        validate_agent_type_and_folder_to_mount(agent_type, folder_to_mount)
+
+        # assign values to class variables
+        self.agent_type = agent_type
+        self.permission = permission
+        self.permission_key = PermissionMapping.MAPPING.get(self.permission)
+        self.system_prompt = get_system_prompt(agent_type, folder_to_mount)
+        self.environment = environment  # TODO: initialize environment
+        self.tools = tools + additional_tools
```

Comment:
`tools` should be based on the agent_type. You can have default set of tools which will be available for all the agents. Additionally, some extra tools for each pre-defined agent.

I think we can ignore it for now as tool class is yet to be implemented.